app/javascript/components/admin/AuditModal.js

```diff
@@ -0,0 +1,338 @@
+      isLoading: false,
+      show: false,
+      all_jurisdiction_paths: {},
+    };
+  }
+
+  hide = () => {
+    this.setState({ isLoading: false }, () => {
+      this.props.onClose();
+    });
+  };
+
+  componentDidMount() {
+    this.updateTable(this.state.query);
+
+    // Gets all jurisdiction paths on initial mount.
+    this.getAllJurisdictionPaths();
+  }
+
+  updateTable = query => {
+    // cancel any previous unfinished requests to prevent race condition inconsistencies
+    this.state.cancelToken.cancel();
+
+    // generate new cancel token for this request
+    const cancelToken = axios.CancelToken.source();
+
+    this.setState({ query, cancelToken, isLoading: true }, () => {
+      this.queryServer(query);
+    });
+  };
+
+  queryServer = _.debounce(query => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .post('/users/audits/' + this.props.user.id, {
+        ...query,
+        cancelToken: this.state.cancelToken.token,
+      })
+      .catch(error => {
+        if (!axios.isCancel(error)) {
+          this.setState(state => {
+            return {
+              table: { ...state.table, rowData: [], totalRows: 0 },
+              isLoading: false,
+            };
+          });
+        } else {
+          console.log(error);
+          this.setState({ isLoading: false });
+        }
+      })
+      .then(response => {
+        if (response && response.data && response.data.audit_rows) {
+          this.setState(state => {
+            const displayedColData = this.state.table.colData.filter(colData => response.data.audit_rows.includes(colData.field));
+            return {
+              table: { ...state.table, displayedColData, rowData: response.data.audit_rows, totalRows: response.data.total },
+              isLoading: false,
+            };
+          });
+        } else {
+          this.setState({ isLoading: false });
+        }
+      });
+  }, 500);
+
+  formatTimestamp(timestamp) {
+    const ts = moment.tz(timestamp, 'UTC');
+    return ts.isValid() ? ts.tz(moment.tz.guess()).format('MM/DD/YYYY HH:mm z') : '';
+  }
+
+  /**
+   * Gets the all possible jurisdictions path via an axios GET request.
+   */
+  getAllJurisdictionPaths() {
+    axios.get('/jurisdictions/allpaths').then(response => {
+      const responseData = response.data.all_jurisdiction_paths;
+
+      // Swap keys and values for ease of use
+      let all_jurisdiction_paths = Object.assign({}, ...Object.entries(responseData).map(([id, path]) => ({ [path]: parseInt(id) })));
+
+      this.setState({ all_jurisdiction_paths });
+    });
+  }
+
+  formatChange = change => {
+    switch (change.name) {
+      case 'locked_at':
+        if (!change.details || !Array.isArray(change.details) || !change.details.length) {
+          return (
+            <span>
+              <b>Account Status</b>: Updated
+            </span>
+          );
+        } else if (change.details[0]) {
```

Comment:
For the true/false audits, I tried providing more context to the user by using languages such as "Locked" v. "Unlocked" and "Enabled" v. "Disabled" so that the meaning of the audit was a little more clear. If we go to this suggestion, I'll have to reword the audit titles to make sense when true/false values are displayed. For example for the section you referenced above, I would probably reword it to display: **Account Locked**: true

So would you like to keep the language displayed in the audits as is or update the language to display true/false for relevant audits and update the title to make the meaning more clear?

Reply:
Yes I would definitely like to keep the language displayed. That was more so just an example of the kind of change I'm suggesting. You can still accomplish this with something like:
```
return (
  <span>
    <b>Account Status</b>: {change.details[0] ? "Unlocked" : "Locked"}
  </span>
);
```

Reply:
Ah, I see that makes sense. Yep, I can look to do that!